test/Table_Tests/src/In_Memory/Table_Format_Spec.enso

```diff
@@ -0,0 +1,394 @@
+            actual.at "bools" . should_equal (table.at "bools")
+            actual.at "ints" . should_equal (table.at "ints")
+            actual.at "floats" . should_equal (table.at "floats")
+            actual.at "strings" . should_equal (table.at "strings")
+            actual.at "date_formats" . should_equal (table.at "date_formats")
+            actual.at "time_formats" . should_equal (table.at "time_formats")
+            actual.at "numeric_formats" . should_equal (table.at "numeric_formats")
+            actual.at "bool_formats" . should_equal (table.at "bool_formats")
+
+        Test.specify "Date and Date_Time, with Date_Time_Formatter" <|
+            expected_dates = Column.from_vector "dates" ["20201221", "20230425"]
+            expected_datetimes = Column.from_vector "datetimes" ["20200110", "20200808"]
+            actual = table.format ["dates", "datetimes"] (Date_Time_Formatter.from "yyyyMMdd")
+            actual.column_names . should_equal table.column_names
+            actual.at "dates" . should_equal expected_dates
+            actual.at "datetimes" . should_equal expected_datetimes
+            actual.at "times" . should_equal (table.at "times")
+            actual.at "bools" . should_equal (table.at "bools")
+            actual.at "ints" . should_equal (table.at "ints")
+            actual.at "floats" . should_equal (table.at "floats")
+            actual.at "strings" . should_equal (table.at "strings")
+            actual.at "date_formats" . should_equal (table.at "date_formats")
+            actual.at "time_formats" . should_equal (table.at "time_formats")
+            actual.at "numeric_formats" . should_equal (table.at "numeric_formats")
+            actual.at "bool_formats" . should_equal (table.at "bool_formats")
+
+        Test.specify "Date_Time and Time_Of_Day, Date_Time_Formatter" <|
+            expected_datetimes = Column.from_vector "datetimes" ["03:04:05", "05:06:07"]
+            expected_times = Column.from_vector "times" ["01:02:03", "10:30:35"]
+            actual = table.format ["datetimes", "times"] (Date_Time_Formatter.from "hh:mm:ss")
+            actual.column_names . should_equal table.column_names
+            actual.at "dates" . should_equal (table.at "dates")
+            actual.at "datetimes" . should_equal expected_datetimes
+            actual.at "times" . should_equal expected_times
+            actual.at "bools" . should_equal (table.at "bools")
+            actual.at "ints" . should_equal (table.at "ints")
+            actual.at "floats" . should_equal (table.at "floats")
+            actual.at "strings" . should_equal (table.at "strings")
+            actual.at "date_formats" . should_equal (table.at "date_formats")
+            actual.at "time_formats" . should_equal (table.at "time_formats")
+            actual.at "numeric_formats" . should_equal (table.at "numeric_formats")
+            actual.at "bool_formats" . should_equal (table.at "bool_formats")
```

Comment:
This is a lot of repetitive text, I'd shorten it into:
```suggestion
            ["bools", "ints", ...].each name-> Test.with_clue name+": " <|
                actual.at name . should_equal (table.at name)
```

Reply:
I think ideally I'd move this into a helper function, eg.:

```
expect_columns_unchanged new_table original_table unchanged_columns =
    unchanged_columns.each col-> Test.with_clue name+": " <|
        new_table.at col . should_equal (original_table.at col)
```

Reply:
Done